### PR TITLE
Migrated KycLevel1 to compose

### LINKED ISSUE #1587: Migrate KYCLevel1Fragment to Jetpack Compose
**Summary:**

Summarize your issue here.

**Steps to reproduce:**

How can we reproduce again the issue?

**Expected behavior:**

What did you expect the app to do?

**Observed behavior:**

What did you see instead?  Describe your issue in detail here.

**Device and Android version:**

What make and model device (e.g., Samsung Galaxy S3) did you encounter this on?  What Android
version (e.g., Android 4.0 Ice Cream Sandwich or Android 6.0 Marshmallow) are you running?  Is it
 the stock
version from the manufacturer or a custom ROM?

**Screenshots:**

Can be created by pressing the Volume Down and Power Button at the same time on Android 4.0 and higher.

### PR DESCRIPTION
**Pull request by AdityaKumdale**
## Issue Fix
Fixes #1587 

## Screenshots

https://github.com/openMF/mobile-wallet/assets/95978829/6d14fa57-c871-4c38-81a4-57e354f18ebb

##

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    ) {
+        OutlinedTextField(
+            value = firstName,
+            onValueChange = {
+                firstName = it },
+            label = {
+                Text(stringResource(R.string.first_name)) },
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(vertical = 7.dp)
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt]: we already have `MfOutlinedTextField ` and `MifosOutlinedTextField` so resuse them

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+                .align(Alignment.CenterHorizontally)
+                .padding(vertical = 7.dp),
+        ) {
+            Text(text = stringResource(R.string.submit))
+        }
+    }
+}
+
+@Preview(showBackground = true)
+@Composable
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt:267]: shouldn't we add preview for kycscreen with all other ui states

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/presenter/KYCLevel1ViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                override fun onError(message: String) {
+                    _kyc1uiState.value = KYCLevel1UiState.Error
+                }
+            }
+        )
+    }
+}
+
+sealed interface KYCLevel1UiState {
+    data object EmptyForm : KYCLevel1UiState
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/presenter/KYCLevel1ViewModel.kt]: rename to empty

**Comment by AdityaKumdale** [comment]
@PratyushSingh07 do we have to build project with gradle 9.0

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]
Rectify this error to solve the failing workflow. You can check this locally by running the `./gradlew detekt`  command in your terminal
image

Also format your code and remove unused imports before pushing

**Comment by PratyushSingh07** [comment]
> @PratyushSingh07 do we have to build project with gradle 9.0

idts

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt (hunk 2 above)
Comment (AdityaKumdale) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt:267]: They are Toasts and screen will navigate back after showing them so I didn't add them, But I think I will add them now. And plus is there any way to preview Togicountrycodepicker , I have researched everywhere but I didn't find any solution yet. Is there any other maintained library available for this.

Comment (PratyushSingh07) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt:267]: I am not sure about ccp. I havent used that in compose, maybe @therajanmaurya would have an idea about some other library

File: core/analytics/src/main/java/org/mifos/mobilewallet/mifospay/core/analytics/AnalyticsEvent.kt [hunk 4]
```diff
@@ -15,10 +15,8 @@ data class AnalyticsEvent(
     val extras: List<Param> = emptyList(),
 ) {
     // Standard analytics types.
-    class Types {
```
Comment (therajanmaurya) [inline core/analytics/src/main/java/org/mifos/mobilewallet/mifospay/core/analytics/AnalyticsEvent.kt]: revert this, I am in middle to build this module.

File: core/analytics/src/main/java/org/mifos/mobilewallet/mifospay/core/analytics/AnalyticsEvent.kt [hunk 5]
```diff
@@ -34,9 +32,7 @@ data class AnalyticsEvent(
     data class Param(val key: String, val value: String)
 
     // Standard parameter keys.
-    class ParamKeys {
```
Comment (therajanmaurya) [inline core/analytics/src/main/java/org/mifos/mobilewallet/mifospay/core/analytics/AnalyticsEvent.kt]: revert this, I am in middle to build this module.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun Kyc1Form(
+    modifier:Modifier,
+    submitData: (String, String,String,String,String,String) -> Unit
+){
+
+    var firstName by remember { mutableStateOf("") }
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt]: change all remember to rememberSavable

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/presenter/KYCLevel1ViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                override fun onError(message: String) {
+                    _kyc1uiState.value = KYCLevel1UiState.Error
+                }
+            }
+        )
+    }
+}
+
+sealed interface KYCLevel1UiState {
+    data object Empty : KYCLevel1UiState
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/presenter/KYCLevel1ViewModel.kt]: Change Empty to `Loading`

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+        navigateBack = navigateBack
+    )
+}
+
+
+@Composable
+fun KYCLevel1Screen(
+    context:Context,
+    uiState: KYCLevel1UiState,
+    submitData: (String, String,String,String,String,String) -> Unit,
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt]: format the code

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+    viewModel: KYCLevel1ViewModel = hiltViewModel(),
+    navigateBack: () -> Unit
+) {
+    val context = LocalContext.current
+    val kyc1uiState by viewModel.kyc1uiState.collectAsStateWithLifecycle()
+
+    KYCLevel1Screen(
+        context = context,
+        uiState = kyc1uiState,
+        submitData = { fname, lname,address1, address2,
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt]: format the code

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+fun KYCLevel1Screen(
+    context:Context,
+    uiState: KYCLevel1UiState,
+    submitData: (String, String,String,String,String,String) -> Unit,
+    navigateBack:() -> Unit
+) {
+
+    when(uiState){
+        KYCLevel1UiState.Empty -> {
+            Kyc1Form(
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt]: Move Kyc1Form from when condition 
```
Kyc1Form(
                modifier = Modifier,
                submitData = submitData
            )

when(uiState){
        KYCLevel1UiState.Loading -> {
             MfOverlayLoadingWheel(stringResource(id = R.string.submitting))
        }

        KYCLevel1UiState.Error -> {
            Toast.makeText(context, stringResource(R.string.error_adding_KYC_Level_1_details),Toast.LENGTH_SHORT).show()
            navigateBack.invoke()
        }

        KYCLevel1UiState.Success -> {
            Toast.makeText(context, stringResource(R.string.successkyc1),Toast.LENGTH_SHORT).show()
            navigateBack.invoke()
        }
    }
 ```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt [hunk 11]
```diff
@@ -0,0 +1,264 @@
...
+            Text(text = stringResource(R.string.submit))
+        }
+    }
+}
+
+@Preview(showBackground = true)
+@Composable
+fun Kyc1FormPreview(){
+    MifosTheme {
+        val context = LocalContext.current
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt]: Don't pass context instead move it KYCLevel1Screen

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt [hunk 12]
```diff
@@ -0,0 +1,264 @@
...
+    )
+}
+
+
+@Composable
+fun KYCLevel1Screen(
+    context:Context,
+    uiState: KYCLevel1UiState,
+    submitData: (String, String,String,String,String,String) -> Unit,
+    navigateBack:() -> Unit
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/kyc/ui/KYCLevel1Screen.kt]: rename to navigateToKycLevel2

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

**Review by therajanmaurya — APPROVED** [review]

**Pull request merged**